include/Server/include/ClientSession/src/SmtpRequest.cpp

```diff
@@ -0,0 +1,163 @@
+#include "SmtpRequest.h"
+#include "MailMessageBuilder.h"
+#include "Base64.h"
+#include "Logger.h"
+
+#include <stdexcept>
+#include <algorithm>
+
+using ISXMM::MailMessageBuilder;
+using ISXBase64::Base64Decode;
+
+namespace ISXSmtpRequest
+{
+SmtpRequest RequestParser::Parse(const std::string& request)
+{
+    SmtpRequest smtp_request;
+    smtp_request.data = request;
+
+    if (request.find("EHLO") == 0)
+    {
+        smtp_request.command = SmtpCommand::EHLO;
+    }
+    else if (request.find("HELP") == 0)
+    {
+        smtp_request.command = SmtpCommand::HELP;
+    }
+    else if (request.find("NOOP") == 0)
+    {
+        smtp_request.command = SmtpCommand::NOOP;
+    }
+    else if (request.find("STARTTLS") == 0)
+    {
+        smtp_request.command = SmtpCommand::STARTTLS;
+    }
+    else if (request.find("REGISTER") == 0)
+    {
+        smtp_request.command = SmtpCommand::REGISTER;
+    }
+    else if (request.find("AUTH") == 0)
+    {
+        smtp_request.command = SmtpCommand::AUTH;
+    }
+    else if (request.find("MAIL FROM") == 0)
+    {
+        smtp_request.command = SmtpCommand::MAILFROM;
+    }
+    else if (request.find("RCPT TO") == 0)
+    {
+        smtp_request.command = SmtpCommand::RCPTTO;
+    }
+    else if (request.find("DATA") == 0)
+    {
+        smtp_request.command = SmtpCommand::DATA;
+    }
+    else if (request.find("RSET") == 0)
+    {
+        smtp_request.command = SmtpCommand::RSET;
+    }
+    else if (request.find("QUIT") == 0)
+    {
+        smtp_request.command = SmtpCommand::QUIT;
+    }
+    else {
+        throw std::runtime_error("Invalid SMTP command.");
+    }
+
+    return smtp_request;
+}
+
+std::string RequestParser::ExtractUsername(std::string auth_data)
+{
+    auth_data.erase(std::remove(auth_data.begin(), auth_data.end(), ' '), auth_data.end());
+    if (!auth_data.empty() && auth_data.front() == '<' && auth_data.back() == '>')
+    {
+        return auth_data.substr(1, auth_data.size() - DELIMITER_OFFSET);
+    }
+
+    return std::string();
+}
+
+std::string RequestParser::ExtractSubject(std::string data)
+{
+    std::size_t start_pos = data.find("Subject: ") + 9;
+    std::size_t end_pos = data.find("\r\n", start_pos);
+    
+    if (start_pos == std::string::npos || end_pos == std::string::npos)
+    {
+        throw std::runtime_error("Subject not found.");
+    }
+
+    return data.substr(start_pos, end_pos - start_pos);
+}
+
+std::string RequestParser::ExtractBody(std::string data)
+{
+    std::size_t start_pos = data.find("\r\n\r\n") + 4;
```

Comment:
\r\n can be put into a constant, for example, static constexpr const char* CRLF = “\r\n”;

The value '4' is a magic number. It would be better to define a constant for clarity.